### DISCUSSION TITLE
Sharing external library between users

### DISCUSSION THREAD

**Discussion by puertocastilla** [immich-app/immich#7780] [Feature Request]
Please, implement the sharing external library between users feature

**Comment by greanmurk** [comment]
Just come to ask for this feature myself and found this discussion. Support request.

I want all users (actually family members) have access to one big "common" library, so it's content will appear on their timelines and memories, and users can use search feature.

Right now the problem is that every library should be owned only by one of users and photo placed in this library available only for this user. I tried to bypass this by creating libraries with the same external path for each user. This almost work, but in this case Immich create copy of cache data for each user and waste a bunch of disk space.

Thinking about possible realization I can imagine two ways. Most obvious is ability to create "no owner" external library, available for all users. Or, more flexible, ablility to set additional users for regular owned external library. Of course finally this is up for devs, if they decide to implement it.

BTW, almost the same request also here #7362.

**Reply by aisbergde** [reply]
I also hope that a user management for external libraries is on the road map. After seeing that external libraries have been moved from users to administrator, I am very optimistic!

**Comment by mikesellt** [comment]
I'm just going to add my thoughts here as well.  I'm moving from Synology to TrueNAS, and as a result am moving my photo library over to Immich.  I have a lot of photos already, and I'd like to be able to add them as an external library, but give all of my users (my wife and kids) view access to that external library while still keeping their own individual libraries for cell phone backup.  I don't see a way to do this currently without making individual external libraries for each user and adding the same photo folder in each of them - basically duplicating the external library for each user.  If that library were duplicated, I don't think all of the face recognition would persist between each library.  If we could just have one external library and have it shared between users, they could help match the faces and whatnot.  

Also, great project so far.  I'm really liking Immich in general.  Thanks.

**Reply by theharshl** [reply]
I have this exact same use case, hoping for this!

**Comment by andreasgerstmayr** [comment]
My use case is a shared samba folder with pictures, which I'd to like to integrate in Immich and share with multiple users, without duplicating all thumbnails and database entries.

I wonder if this functionality could replace partner sharing, because sharing a users' upload library has the same effect as partner sharing?

**Reply by soana** [reply]
If a feature like this is implemented can we then export albums from the personal library into the external library - effectively moving assets from the local to external location on the server?

**Reply by andreasgerstmayr** [reply]
I think these are different use cases - the one here is about sharing the external library, the one you are describing is about copying assets from one library to another. I'd suggest to open a new feature request for the feature you're describing.

**Comment by codihuston** [comment]
I would also like this feature, and am considering contributing. Does anyone know what limitations or design philosophies currently exist in the code base that prevent this from happening?

**Reply by andreasgerstmayr** [reply]
I think it gets tricky regarding
* should the timeline show pictures of all libraries by default, or make it customizable? (afaics for partner sharing, it's disabled by default but there is an option to view partners' pictures on the timeline)
* same for map markers
* what about tagged persons - if and how they should be merged with the tagged persons of the current user (see https://github.com/immich-app/immich/discussions/7038 for a related discussion)

**Comment by mmomjian** [comment]
Duplicate of #7362 as mentioned above, thank you.

**Comment by embdMan** [comment]
Yes, please! I need it! 
I went all the way, installing it, importing external libraries and so on... unluckily the last thing I did is - at the end - create the user for my wife and now... Very disappointed, didn't see this coming.
Any way we can support this feature request? Need a helping hand?

**Comment by daudo** [comment]
I am facing the same dilemma and yes, shareable external libraries would be a very valuable feature for immich.

However, as some kind of dirty workaround, I have created a separate "family" user who owns the external libraries. This "family" user has enabled partner sharing for my family and me, giving us all access to our legacy shared photos. This doesn't really scale, but it's good enough for me and my family.

**Reply by corsac-s** [reply]
I tried to do the same but unfortunately in my case there are (a lot of) duplicates between external libraries for different users and deduplication doesn't work accross partner sharing.

And trying to add the same path to multiple user-owned external libraries means I'll have duplicated thumbnails (since they're indexed by asset id which doesn't seem linked to the filepath at all)

**Comment by higheraims** [comment]
I came here looking for exactly this. I'm running the app on TrueNAS and I have a large existing library that I have linked as an external library. Now I want to share it with other users.

**Reply by DerSchnelle-noob** [reply]
Same here, external library with over 400k photos that I would like to share.

**Comment by realrolfje** [comment]
I haven't tried (and it would mean you have to do everything on an external library outside immich), but one workaround for now would be to add all photos to an external library and then share that folder on every user. It has big downsides though:
- Added metadata is not shared
- will probably cause duplicate storage

For me the case is the same: I have over 20 years of photos on a server, in folders. I'd like to make these available to the family through immich, to make things fast and simple for them. I am also looking for a way for new photo's which are dropped in immich to migrate back to the folder structure, as I have seen many photo management tools come and go in the last 20 years. That is why I always opt for structuring photo's in such a way that it would never break if a tool dissappears.

Maybe Immich could have a "folder viewer" feature where you can select a read-only folder that is available to all (or selected) users of immich.

**Comment by drewmerc302** [comment]
I just spent 45 minutes trying to figure out why my wife couldn't see my external library on her user account even after i enabled partner sharing. I just stumbled across this thread. For me, Immich is essentially worthless without this feature and i certainly wouldn't have spent the better part of a week installing, configuring, and letting import jobs run on 200,000+ photos/videos if i had known the only person that could see them would be me.

+1 on the ability to share an existing external library via partner sharing.

**Reply by arnschi** [reply]
But you can do this.
In the admin account, you set up partner sharing.
In the partner's account, go to the partner sharing settings. Here the "admin partner" already appears, and you can activate that you see the partner's pictures in the timeline.

**Reply by arnschi** [reply]
OK got it. The photos appear in the timeline, but metadata like people's names don't show up. This is too bad.

**Reply by drewmerc302** [reply]
Wow... i'm going to assume i missed the detail about having to enable timeline view in the shared partners partner sharing submenu even after adding a shared partner as the admin user. I spent quite a while reading official documentation, reddit posts, and trolling thru half a dozen github threads trying to figure it all out. 

this is good enough for me, thank you kind sir/ma'am!!!

**Comment by realrolfje** [comment]
I have tried the partner sharing but that makes it no better than any other app, as it only shows in the timeline (not sure why that is). For now I switched to Jellyfin. Also not ideal, but at least that enables everybody in the house to see the pictures on our shared drive. I may use Immich for tagging and exporting because I would really like to be able to use the face recognition and location on the map. Immich is pretty good at that.

**Comment by pkoevesdi** [comment]
Same here, so disappointing, after all the setup, that my family members can't see the external library.  Now I didn't even find a way to make them self have an external library? Is that gone? In general, only an admin can have an external library?

**Reply by pkoevesdi** [reply]
Ok, at least I figured out, that the admin can share the same folder as a new external library for each user. Still, everything gets scanned and recognized again for each user, not to mention that the metadata (persons etc.) is not shared. I'd suggest to put this feature very far up on prios, that would be nice. Does it make sense to make a bounty for it? How can I?

**Reply by realrolfje** [reply]
Jellyfin makes this possible, but does not have the cool face recognition feature. The only way to share an external library is to have 1 account and share that password with the family :-(

**Comment by RobinEd** [comment]
Ummmm... really this is not possible? That can't be true, right?

**Reply by realrolfje** [reply]
Sadly, this is true as far as my extended experiments go. If you think this is possible then please share with us how to share an external library between two users. It would solve a problem for quite a few of us. 👍

**Comment by wintbiit** [comment]
up. we really need this!!!!!!!!!!

**Comment by broman-pl** [comment]
Two observations 
- admin user sees other users external library in their timeline :worried: 
- scan of one external library triggers scan of other lib as well :worried: 

And a feature request, it will be nice to also have option to say what portion of external library we don't like to share :smile:

**Comment by OTheNonE** [comment]
I structure my photos manually in folders directly on my drive and map them into Immich via "External Library", and use Immich only for viewing. Therefore, my prefered way of looking through photos in Immich is via the "Folders" viewer, where i can browse through my drives folder structure. Currently, eventhough the photos are viewable in the timeline viewer for other users, Folders view for one External Library are only viewable for the owner of the library and not for anyone else. There is no option for sharing the external library with other users, so that they can view the photos via the Folder viewer.

So... highly desired feature!
